src/_alfasim_sdk/alfacase/case_description.py

```diff
@@ -1991,3 +1992,52 @@ def reset_invalid_references(self):
         self._check_pvt_model_files(reset_invalid_reference=True)
         self._check_pvt_model_references(reset_invalid_reference=True)
         self._check_fluid_references(reset_invalid_reference=True)
+
+    def ensure_unique_name(self):
+        """
+        Ensure that elements that can be referenced by name have a unique name.
+        """
+        import collections
+        from collections import Counter
+
+        duplicate_names = collections.defaultdict(list)
+
+        def get_duplicate_keys(counter):
+            return [key for key, value in counter.items() if value > 1]
+
+        all_node_names = [i.name for i in self.nodes]
+        all_pipes_names = [i.name for i in self.pipes]
+        all_wells_names = [i.name for i in self.wells]
+        all_walls_names = [i.name for i in self.walls]
+        all_material_names = [i.name for i in self.materials]
+        all_pvt_names = list(self.pvt_models.tables.keys())
+        all_pvt_names += list(self.pvt_models.correlations.keys())
+        all_pvt_names += list(self.pvt_models.compositions.keys())
+
+        all_fluids = [
+            fluid
+            for pvt_model in self.pvt_models.compositions.values()
+            for fluid in pvt_model.fluids.keys()
+        ]
+
+        duplicate_names["Nodes"] = get_duplicate_keys(Counter(all_node_names))
+        duplicate_names["Pipes"] = get_duplicate_keys(Counter(all_pipes_names))
+        duplicate_names["Wells"] = get_duplicate_keys(Counter(all_wells_names))
+        duplicate_names["Walls"] = get_duplicate_keys(Counter(all_walls_names))
+        duplicate_names["Materials"] = get_duplicate_keys(Counter(all_material_names))
+        duplicate_names["PVT"] = get_duplicate_keys(Counter(all_pvt_names))
+        duplicate_names["Fluids"] = get_duplicate_keys(Counter(all_fluids))
+
+        def get_error_msg():
+            output = []
+            for key, value in duplicate_names.items():
+                output.append(f"{key}:")
+                formatted_names = "\n    - ".join(name for name in value)
+                output.append(f"    - {formatted_names}")
+            return "\n".join(output)
+
+        if any(value for key, value in duplicate_names.items()):
```

Comment:
```suggestion
        if any(value for key, value in sorted(duplicate_names.items())):
```

Reply:
Sorry I didn't understand what the benefit of sorting the list here, I'm just checking if any of the entries have a list with elements
```python
>>>  d1 = {'a': []}
>>> d2 = {'a': [], 'b': [1]}
>>> any(value for key, value in (d1.items())
False
>>> any(value for key, value in d2.items())
True
```

Reply:
Oh you are right, I meant sorting the dict in `get_error_msg`. Updating.